src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -108,9 +116,85 @@ public long RefreshIn
 
         public HttpResponse HttpResponse { get; set; }
 
+        public new MsalTokenResponse DeserializeFromJson(string json)
```

Comment:
@bgavrilMS This method shows the added complexity well. A ton of boilerplate code. Has to be this `(TokenSource)Convert.ToInt32(jObject[TokenSourcePropertyName]?.ToString(), CultureInfo.InvariantCulture);`, instead of simpler `jObject.Value<TokenSource>("TokenSource")`. Also serializing `HttpResponse` and its inner properties like `Headers` becomes painfully impractical.

Reply:
Would it be code-generable?

Reply:
> Would it be code-generable

I think current approach is fine, but https://github.com/neuecc/Utf8Json code gen was suggested in

https://github.com/dotnet/runtimelab/issues/635#issuecomment-773292026

Reply:
Thanks for following this @charlesroddie . Yeah, we generally can't take dependencies in MSAL, as it causes dependency hell. 
@pmaytak - we discussed this offline, so let's see if Unity suggested solution would work. But as a side note, HttpResponse isn't serialized, it's only assigned to (and should be refactored out of there!)